zebra-chain/src/orchard/shielded_data.rs

```diff
@@ -0,0 +1,59 @@
+//! Orchard shielded data for `V5` `Transaction`s.
+
+use crate::{
+    amount::Amount,
+    orchard::{tree, Action},
+    primitives::{
+        redpallas::{Binding, Signature, SpendAuth},
+        Halo2Proof,
+    },
+    serialization::AtLeastOne,
+};
+
+/// A bundle of [`Action`] descriptions and signature data.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct ShieldedData {
+    /// The Orchard Actions.
+    pub actions: AtLeastOne<AuthorizedAction>,
+    /// The orchard flags for this transaction.
+    pub flags: Flags,
+    /// The net value of Orchard spends minus outputs.
+    pub value_balance: Amount,
+    /// The shared anchor for all `Spend`s in this transaction.
+    pub shared_anchor: tree::Root,
+    /// The aggregated zk-SNARK proof.
+    pub proof: Halo2Proof,
+    /// A signature on the transaction hash.
```

Comment:
Clarify: Which transaction hash, there are 3? Non-malleable id, auth data hash, and sighash.
```suggestion
    /// A signature on the transaction `sighash`.
```